mesonbuild/dependencies/ui.py

```diff
@@ -493,3 +493,56 @@ def check_wxconfig(self):
                 pass
         WxDependency.wxconfig_found = False
         mlog.log('Found wx-config:', mlog.red('NO'))
+        
+class VulkanDependency(ExternalDependency):
+    def __init__(self, environment, kwargs):
+        super().__init__('vulkan', environment, None, kwargs)
+        
+        if DependencyMethods.SYSTEM in self.methods:
+            try:
+                self.vulkan_sdk = os.environ['VULKAN_SDK']
+                if not os.path.isabs(self.vulkan_sdk):
+                    raise DependencyException('VULKAN_SDK must be an absolute path.')
+                    
+                # TODO: this config might not work on some platforms, fix bugs as reported
+                # we at least have to detect other 64-bit platforms (e.g. armv8)
+                lib_name = 'vulkan'
```

Comment:
The so major version should be explicit. When at some point in the future Vulkan 2.0 is released with an incompatible loader, `libvulkan.so.2`, the current code will wrongly choose the Vulkan 2.0 loader if it's installed on the target platform.

Using `compiler.find_library` is fine, I have no comments on that part. :)